playground/src/kata.tsx

```diff
@@ -4,14 +4,56 @@
 import { useEffect, useRef } from "preact/hooks";
 import {
   CompilerState,
+  ExplainedSolution,
   ICompilerWorker,
   ILanguageServiceWorker,
   Kata,
+  Lesson,
   QscEventTarget,
+  Question,
 } from "qsharp";
 import { Editor } from "./editor.js";
 import { OutputTabs } from "./tabs.js";
 
+function ExplainedSolutionAsHtml(solution: ExplainedSolution): string {
+  let html = "";
+  for (const item of solution.items) {
+    if (item.type === "example" || item.type === "solution") {
+      html += `<pre><code>${item.code}</code></pre>`;
+    } else if (item.type === "text-content") {
```

Comment:
Is this an exhaustive list of 'types'? Should there be an 'else throw Err' in case you missed one? (Or use a `switch` statement and use exhaustive type checking)

Reply:
Using `switch` and exhaustive type checking now.